packages/ui/src/components/Pagination/SPagination.vue

```diff
@@ -0,0 +1,366 @@
+    current.value === page ||
+    page === 1 ||
+    page === pagesNum.value ||
+    (!isAbleJumpNext.value && page >= pagesNum.value - PAGINATION_JUMP_SIZE) ||
+    (!isAbleJumpPrev.value && page <= PAGINATION_JUMP_SIZE + 1) ||
+    (isAbleJumpNext.value && isAbleJumpPrev.value && Math.abs(current.value - page) < PAGINATION_JUMP_SIZE / 2)
+  )
+}
+
+function handleJumpClick(value: PaginationJumpValues) {
+  if (value === 'jumpNext') {
+    if (isAbleJumpNext.value) {
+      current.value = Math.min(pagesNum.value, current.value + PAGINATION_JUMP_SIZE)
+    }
+
+    return
+  }
+
+  if (value === 'jumpPrev') {
+    if (isAbleJumpPrev.value) {
+      current.value = Math.max(0, current.value - PAGINATION_JUMP_SIZE)
+    }
+
+    return
+  }
+
+  current.value = value
+}
+
+function handleNextClick() {
+  if (isCurrentLast.value) {
+    return
+  }
+
+  const newPage = current.value + 1
+  emit('click:next', newPage)
+  current.value = newPage
+}
+
+function handlePrevClick() {
+  if (isCurrentFirst.value) {
+    return
+  }
+
+  const newPage = current.value - 1
+  emit('click:prev', newPage)
+  current.value = newPage
+}
+</script>
+
+<template>
+  <div
+    ref="pagination"
+    class="s-pagination flex flex-wrap justify-between flex-row-reverse"
+    data-testid="pagination"
+  >
+    <div
+      class="order-last flex"
+      :class="{
+        'basis-full justify-center mt-16px': shouldWrap,
+      }"
+    >
+      <div
+        ref="paginationLeftPanel"
+        class="flex"
+      >
+        <div
+          class="s-pagination__count whitespace-nowrap sora-tpg-p4 mr-32px flex items-center"
+          data-testid="pagination-progress"
+        >
+          <slot
+            name="progress"
+            v-bind="{ firstItemNum, lastItemNum, total }"
+          >
+            {{ firstItemNum }}—{{ lastItemNum }} of {{ total }}
+          </slot>
+        </div>
+
+        <div class="s-pagination__sizes whitespace-nowrap flex items-center">
+          <div class="s-pagination__sizes-label sora-tpg-p4 mr-16px">
+            {{ sizesLabel }}
+          </div>
+          <SDropdown
+            v-model="numericSize"
+            size="sm"
+            :options="sizeOptions"
+          />
+        </div>
+      </div>
+    </div>
+
+    <div
+      class="order-first flex"
+      :class="{
+        'basis-full justify-center': shouldWrap,
+      }"
+    >
+      <div
+        ref="paginationRightPanel"
+        class="flex"
+      >
+        <div class="s-pagination__pages flex items-center">
+          <template
+            v-for="pageButton in pageButtons"
+            :key="pageButton"
+          >
+            <button
+              class="s-pagination__button flex justify-center w-24px sora-tpg-h7"
+              :class="{
+                's-pagination__button_active': current === pageButton,
+                'ml-4px': pageButton > 1,
+              }"
+              :data-testid="current === pageButton ? 'active-button' : 'page-button'"
```

Comment:
I think it's not good to use *any* not trivial logic for `testid` selectors. We need to test component production behavior, but not what it exposes purely for tests. In this case, I think it's better to target `s-pagination__button` and `s-pagination__button_active` classes instead of variative `testid`.